test/xpu_api/asynch.pass.cpp

```diff
@@ -0,0 +1,142 @@
+#include "oneapi/dpl/iterator"
+
+#include "support/pstl_test_config.h"
+
+#include <iostream>
+#include <iomanip>
+
+#if TEST_SYCL_PRESENT
+#    include <CL/sycl.hpp>
+#endif
+
+template <typename _T1, typename _T2>
+void
+ASSERT_EQUAL(_T1&& X, _T2&& Y)
+{
+    if (X != Y)
+        std::cout << "CHECK CORRECTNESS (ASYNC): fail (" << X << "," << Y << ")" << std::endl;
+}
+
+#if TEST_DPCPP_BACKEND_PRESENT
+void
+test_with_buffers()
+{
+    const int n = 100;
+    {
+        sycl::queue q;
+        sycl::buffer<int> x{n};
+        sycl::buffer<int> y{n};
+
+        auto my_policy = oneapi::dpl::execution::make_device_policy(q);
+        auto res_1a = oneapi::dpl::experimental::copy_async(my_policy, oneapi::dpl::counting_iterator<int>(0),
+                                                            oneapi::dpl::counting_iterator<int>(n),
+                                                            oneapi::dpl::begin(x)); // x = [0..n]
+        auto res_1b = oneapi::dpl::experimental::fill_async(my_policy, oneapi::dpl::begin(y), oneapi::dpl::end(y),
+                                                            7); // y = [7..7]
+
+        auto res_2a = oneapi::dpl::experimental::for_each_async(
+            my_policy, oneapi::dpl::begin(x), oneapi::dpl::end(x), [](int& e) { ++e; }, res_1a); // x = [1..n]
+        auto res_2b = oneapi::dpl::experimental::transform_async(
+            my_policy, oneapi::dpl::begin(y), oneapi::dpl::end(y), oneapi::dpl::begin(y),
+            [](const int& e) { return e / 2; },
+            res_1b); // y = [3..3]
+
+        sycl::buffer<int> z{n}; //std::vector<int> z(n);
+        auto res_3 = oneapi::dpl::experimental::transform_async(my_policy, oneapi::dpl::begin(x), oneapi::dpl::end(x),
+                                                                oneapi::dpl::begin(y), oneapi::dpl::begin(z),
+                                                                std::plus<int>(), res_2a, res_2b); // z = [4..n+3]
+
+        auto alpha = oneapi::dpl::experimental::reduce_async(my_policy, oneapi::dpl::begin(x), oneapi::dpl::end(x), 0,
+                                                             std::plus<int>(),
+                                                             res_2a)
+                         .get(); // alpha = n*(n+1)/2
+
+        auto beta =
+            oneapi::dpl::experimental::transform_reduce_async(my_policy, oneapi::dpl::begin(z), oneapi::dpl::end(z), 0,
+                                                              std::plus<int>(), [=](int e) { return alpha * e; })
+                .get();
+
+        ASSERT_EQUAL(beta, (n * (n + 1) / 2) * ((n + 3) * (n + 4) / 2 - 6));
+    }
+}
+
+void
+test_with_usm()
+{
+    cl::sycl::queue q;
+    const int n = 13;
+
+    // ASYNC TEST USING USM //
+
+    {
+        // Allocate space for data using USM.
+        uint64_t* data1 =
+            static_cast<uint64_t*>(cl::sycl::malloc_shared(n * sizeof(uint64_t), q.get_device(), q.get_context()));
```

Comment:
whether don't we have USM utilities in C++ style?
Like
`auto  p = usm_alloc<T>(count, q);`

Reply:
`malloc_shared` is widely used across our set of USM tests. We might want to consider modernizing all of our tests. I'll leave it for now.